tests/storages_tests/test_cached_storage.py

```diff
@@ -28,6 +28,19 @@ def test_create_trial() -> None:
     storage.create_new_trial(study_id)
 
 
+def test_set_trial_state() -> None:
+    base_storage = RDBStorage("sqlite:///:memory:")
+    storage = _CachedStorage(base_storage)
+    study_id = storage.create_new_study("test-study")
+    trial_id = storage.create_new_trial(study_id)
+    storage.set_trial_state(trial_id, TrialState.COMPLETE)
+
+    cached_trial = storage.get_trial(trial_id)
+    base_trial = storage._backend.get_trial(trial_id)
```

Comment:
This is definitely not a strong opinion but instead of accessing a private attribute, I think you can `load_study` to create a new study with a cleared cache.

Reply:
It's not your suggested approach, but I updated the PR to use `base_storage.get_trial(trial_id)` for removing accessing to a private attribute.